tensorflow/core/platform/default/logging.cc

```diff
@@ -366,7 +366,7 @@ bool LogEveryNState::ShouldLog(int n) {
 
 bool LogFirstNState::ShouldLog(int n) {
   const uint32 counter_value = counter_.load(std::memory_order_relaxed);
-  if (counter_value < n) {
+  if (counter_value < (size_t)n) {
```

Comment:
If `n` is negative `static_cast<size_t>(n)` (using modern casts) is a very large number. Instead, let's make `counter_value` be an `int` and do a static cast there on the RHS of line 368

Reply:
`const int counter_value = static_cast<size_t>( counter_.load(std::memory_order_relaxed) );`

Is the above what you were suggesting?

Reply:
`static_cast<int>`, but yes